### ISSUE TITLE
Deadline: Optimize environment injection (GlobalJobPreLoad)

### ISSUE THREAD

**Issue by BigRoy** [pypeclub/OpenPype#2592]
### Issue

The Deadline integration uses the [`GlobalJobPreLoad.py`](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/plugins/GlobalJobPreLoad.py) to [inject the OpenPype environment](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/plugins/GlobalJobPreLoad.py#L11-L92) for the render machines for a specific job/task.

The problem is running the OpenPype executable to extract these environments on our machines **takes anywhere between 15-25 seconds per task** (the 15 seconds is best case scenario with the openpype executable running locally). This overhead means that for a render job where frames take about a minute to render we're looking at 25% OpenPype overhead to each frame.

With 15 seconds overhead per frame it means that just for **1000 frames (at single frame per task) we are looking at 15000 seconds which equals roughly a _whopping 4.16 hours_ overhead.** (Somehow I keep feeling that doesn't sound right; did I calculate that right?)

I also raised [this issue on Discord](https://discord.com/channels/517362899170230292/517382145552154634/935856130482143242) so referencing it here.

#### Don't go through OpenPype Executable

Note: The OpenPype executable currently does some version resolution to figure out which version of OpenPype to actually add to the current environment (as [mentioned on Discord](https://discord.com/channels/517362899170230292/517382145552154634/935873941111734273)) to trigger the code in which ensures you call `extractenvironments` through the right OpenPype version. So going as "direct" as the code snippet here might not be feasible for production since it lacks the version management.

There are consistency reasons with regards to dependencies for OpenPype (as @jakubjezek001 [mentioned on discord](https://discord.com/channels/517362899170230292/517382145552154634/935858167714283570)) to run the `extractenvironments` through the OpenPype executable (especially also with regard to potential non standard lib / built-in Python libraries to run OpenPype API - but [collecting the same environment with `get_app_environments_for_context`](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/openpype/lib/applications.py#L1150-L1209) takes about 0.1 second.
```python
import time
import openpype.lib.applications

fn = openpype.lib.applications.get_app_environments_for_context

s = time.time()
fn(project_name="sandbox", 
   asset_name="debug",
   task_name="generic",
   app_name="maya/2022",
   env_group="farm")
e = time.time()
print(e-s)
# 0.10447096824645996
```

<details>
  <summary>Click to show alternative timings</summary>

***Alternative Timing***
```python
import timeit

setup = 'from openpype.lib.applications import get_app_environments_for_context'
cmd = 'get_app_environments_for_context(project_name="sandbox", asset_name="debug", task_name="generic", app_name="maya/2022", env_group="farm")'

repeat = 1
t = timeit.timeit(setup=setup, stmt=cmd, number=repeat)
print(t / float(repeat))
# 0.10154770000008284
```

</details>

That's a major performance difference for something so basic. I'd say it's quite worth exposing a "minimal" OpenPype API that potentially uses standard library only (or has very few dependencies) just for these cases. That would be the best case scenario really. The GlobalJobPreLoad should be running through a sandboxed Python that comes with Deadline - and thus the environment it runs it should be (fairly?) consistent across render machines?

---

#### Call once per Deadline Job for a machine?

Currently the `GlobalJobPreLoad` also seems to trigger for every single task of a Job on our end, even for the same machine rendering multiple tasks on the same job after each other. It could be worth looking into settings / options to bring that back to a single time per Job since all the tasks in the single job would share the same environment (in case of MayaBatch they should even be sharing the same process!) 

I'm not sure why it's doing this. Looking back at older Deadline logs we've used GlobalJobPreLoad before (outside of OpenPype) and it triggered solely for the first task the machine rendered consecutively on the job. If it would render another job (e.g. higher priority job comes in between) and come back then it would trigger it again.

Having this run once per Job only would at least reduce the overhead to not trigger every Task - for fast renders with many tasks (e.g. higher number of frames) that could be well worth it.

I mean with a 1000 task job already mean reducing that four hour overhead by a lot (depending on number of unique machines, etc.)

---

#### Optimize how fast we can trigger extractenvironments through OP executable

We might want to look into our options to have OpenPype start as fast as possible to solely access the API methods like `extractenvironments`.

[cuID:OP-2519]

**Comment by BigRoy** [comment]
> I'm not sure why it's doing this. Looking back at older Deadline logs we've used GlobalJobPreLoad before (outside of OpenPype) and it triggered solely for the first task the machine rendered consecutively on the job. If it would render another job (e.g. higher priority job comes in between) and come back then it would trigger it again.

Maybe the GlobalJobPreLoad triggering again for each task with the same machine might be due to how we explicitly invalidate the cache with [https://docs.thinkboxsoftware.com/products/deadline/10.1/2_Scripting%20Reference/class_deadline_1_1_scripting_1_1_repository_utils.html#af3134d482f11af84fea8fa888851321a](`RepositoryUtils.GetJob()`) in the `GlobalJobPreLoad.py` [here](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/plugins/GlobalJobPreLoad.py#L186) and [here](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/plugins/GlobalJobPreLoad.py#L17)? It's the only real different methods OpenPype's GlobalJobPreLoad is doing compared to our older one.

**Comment by iLLiCiTiT** [comment]
Just to put the performance timing into context. When you call the `get_app_environments_for_context` function timing you probably already have imported openpype with it's dependencies. So mongo connection is alive in global scope and does not need to create new connection and settings are probably reused from cache (cache of settings is cleared each 10 seconds). It will be always faster in process when OpenPype was already imported. When you need to do all of that from scratch you probably won't get much faster results no matter if you spawn new process or hack it to import all of that in deadline plugin. So for me the main question is what takes so long to be able find out if we can skip some parts when we need to get environments...

**Comment by BigRoy** [comment]
> Just to put the performance timing into context. When you call the get_app_environments_for_context function timing you probably already have imported openpype with it's dependencies. So mongo connection is alive in global scope and does not need to create new connection and settings are probably reused from cache (cache of settings is cleared each 10 seconds). It will be always faster in process when OpenPype was already imported. When you need to do all of that from scratch you probably won't get much faster results no matter if you spawn new process or hack it to import all of that in deadline plugin.

Correct - going from a Python 3.7.9 shell the timing results in roughly 0.56 seconds.

**Comment by kalisp** [comment]
I think invalidation of cache is there for injection actually work at all, without it it ignored passing env var values through.

**Comment by BigRoy** [comment]
> I think invalidation of cache is there for injection actually work at all, without it it ignored passing env var values through.

In our old pipeline we didn't have the code and that worked fine - there all we did was
```python
deadlinePlugin.SetProcessEnvironmentVariable(key, value)
```

And we got the input environment variables either directly from `os.environ` or `deadlinePlugin.GetProcessEnvironmentVariable(key)`.

So I'm not sure if what you're saying is correct since it worked for us originally too.

We do have this Deadline setting to Run Plugin Sandbox in Job Environment enabled - which is also the default. I can imagine having that disabled might influence things however.

![deadline_run_plugin_sandbox_in_job_environment](https://user-images.githubusercontent.com/2439881/151171431-7a652d35-6b59-4053-846c-b8594df14cac.png)

**Comment by iLLiCiTiT** [comment]
> Correct - going from a Python 3.7.9 shell the timing results in roughly 0.57 seconds 1.05 seconds.

So why is it? Is it because the initialization in executable is slow or the logic inside openpype is slow? If you run the same `extractenvironments` command from terminal does it take the same time? Is it different per workstation? We have to find what is the slow part first.

I think that we have to be able to deal with farms where are crazy combinations of workstations like centos, ubuntu and windows where ubuntu don't have the same paths as centos machines because have to use different executables.

**Comment by kalisp** [comment]
> In our old pipeline we didn't have the code and that worked fine - there all we did was
Interesting, I will double check it again, but I vaguely remember I was quite confused and angry without the invalidation, but maybe I messed up something.

**Comment by BigRoy** [comment]
> I think that we have to be able to deal with farms where are crazy combinations of workstations like centos, ubuntu and windows where ubuntu don't have the same paths as centos machines because have to use different executables.

Agreed. There are some good arguments to still have it go through something precompiled like the console.

Here's a speed test going through Powershell measuring time taken by an executable:

**Powershell**
```powershell
Measure-Command { ./openpype_console.exe extractenvironments S:/env.json --project "sandbox" --asset "debug" --task "generic" --app "maya/2022" --envgroup "farm" | Out-Default }
```

**Output**
```
>>> run disk mapping command ...
*** Cannot get OpenPype path from database.
Finding latest version
>>> Injecting OpenPype version to running environment  ...
>>> Validating version [ C:\Users\Roy\AppData\Local\pypeclub\openpype\openpype-v3.8.0-nightly.3.1 ]
--- version is valid
>>> loading environments ...
  - for Avalon ...
  - global OpenPype ...
  - for modules ...


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 2
Milliseconds      : 148
Ticks             : 21487937
TotalDays         : 2.48702974537037E-05
TotalHours        : 0.000596887138888889
TotalMinutes      : 0.0358132283333333
TotalSeconds      : 2.1487937
TotalMilliseconds : 2148.7937
```

This was directly using the OpenPype executable that was in my `build/`
That's much much faster. :/

**Comment by BigRoy** [comment]
> I'm not sure why it's doing this. Looking back at older Deadline logs we've used GlobalJobPreLoad before (outside of OpenPype) and it triggered solely for the first task the machine rendered consecutively on the job. If it would render another job (e.g. higher priority job comes in between) and come back then it would trigger it again.

@kalisp I do notice now that the consecutive task renders show this message:

> {timestamp}:  0: Plugin will be reloaded: Plugin files are out of date and must be reloaded.

Not sure why it's saying they are outdated every time - but likely this is related?

**Comment by antirotor** [comment]
> (Not sure why it's still "validating version" but not much speed difference either)

Side note: validation is step where content of `checksums` file is compared against list of files and their checksums to validate version integrity. If there is no checksum file, it is skipped. Still I can't believe it is able to calculate those sha256 hashes from files and compare them against the list so fast.

**Comment by BigRoy** [comment]
> Side note: validation is step where content of checksums file is compared against list of files and their checksums to validate version integrity. If there is no checksum file, it is skipped. Still I can't believe it is able to calculate those sha256 hashes from files and compare them against the list so fast.

Enabling `OPENPYPE_DONT_VALIDATE_VERSION` didn't change much either.

**Powershell**
```powershell
$env:OPENPYPE_DONT_VALIDATE_VERSION = "1"
$env:OPENPYPE_HEADLESS_MODE = "1"
Measure-Command { ./openpype_console.exe extractenvironments S:/env.json --project "test" --asset "asset1" --task "debug" --app "maya/2022" --envgroup "farm" | Out-Default }
```

**Result:**
```
>>> run disk mapping command ...
Finding studio version "3.8.0-nightly.5"
>>> Injecting OpenPype version to running environment  ...
>>> Validating version [ C:\Users\Roy\AppData\Local\pypeclub\openpype\openpype-v3.8.0-nightly.5 ]
--- version is valid
>>> loading environments ...
  - for Avalon ...
  - global OpenPype ...
  - for modules ...


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 13
Milliseconds      : 760
Ticks             : 137605341
TotalDays         : 0.000159265440972222
TotalHours        : 0.00382237058333333
TotalMinutes      : 0.229342235
TotalSeconds      : 13.7605341
TotalMilliseconds : 13760.5341
```

**Comment by BigRoy** [comment]
When Right clicking the Job -> **View Job History** it shows a lot of **Saved Job** logs from the machines that are rendering. I felt that definitely must have been related. And it seemed that it would trigger the `GlobalJobPreLoad` whenever Saved Job had triggered.

Whenever you Right Click a job and "Modify Job Properties" click through menu items on the left hand side and click Cancel without changing properties of the Job a "Save Job" is also triggered (see it getting added when doing "Show History" again) and Deadline feels the need to reload the Plug-in. Which I guess is a safe bet.

But why are all the machines triggering "Saves" of the job?

---

Somehow I have the feeling it must be the [Event Plug-in](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/events/OpenPype/OpenPype.py) maybe trying to set the `openpype_executable` again and again. :) 

Upon disabling the OpenPype Event Plug-in in Configure and Resubmitting the Job directly in Deadline (to ensure it really restarts and doesn't use anything from cache) the machines don't register "Save Job" anymore.

Could it be [OnJobStarted](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/events/OpenPype/OpenPype.py#L114) or [OnSlaveStartingJob](https://github.com/pypeclub/OpenPype/blob/fdcbd03d90e2e64ed80ca937cc4ed758a3dc642e/vendor/deadline/custom/events/OpenPype/OpenPype.py#L176) triggering for each machine, it updating the job... and messing with the other machines?

**Comment by BigRoy** [comment]
So with the Event disabled and all code in the event disabled - it basically will not do anything - the issue has been reduced but not gone away. It does **not** save Jobs anymore - so I guess that's cleaned. However every few minutes when rendering the same Job it still does the same - making me assume the issue is still not (completely?) fixed.

---

This [Deadline support topic](https://forums.thinkboxsoftware.com/t/3ds-max-exits-restarts-between-every-frame-checkbox-not-selected/13438/4) confirms my suspicion that this line in the log indeed describes why Deadline is running `GlobalJobPreLoad` again:
> Plugin will be reloaded: Plugin files are out of date and must be reloaded.

Later in the topic Deadline support also describes [what might be happening](https://forums.thinkboxsoftware.com/t/3ds-max-exits-restarts-between-every-frame-checkbox-not-selected/13438/6). 

It could be that Deadline is detecting changes in local cache/temp folders that it's tracking and thus reporting it's not in sync. Something to investigate at least! (Could it be something the GlobalJobPreLoad file is still doing? Maybe the extracted environment? No that must not be it; because it works fine if a single machine renders the full job in isolation.) Could OpenPype be doing something else at X intervals? every 60 seconds or so?

A better question at this stage might actually be if anyone else is able to reproduce the issue?

---

For sake of debugging I've even resynced the Deadline server hardware clock - hehe. It's driving me nuts!

**Comment by BigRoy** [comment]
Seems I found the last culprit that still forced the Plug-in restart every X seconds - which I could actually identify rendering on a single machine too. This time looking at a live Worker log which shows some info that isn't logged into the Task logs.
```
2022-01-26 22:47:19:  All job files are already synchronized
2022-01-26 22:47:19:  Synchronizing Plugin MayaBatch from N:\DeadlineRepository10\plugins\MayaBatch took: 0 seconds
```
These two lines show up additionally each time whenever a new task kicks in with running `GlobalJobPreLoad` again:
```
2022-01-26 22:47:19:  0: Plugin will be reloaded: Plugin files are out of date and must be reloaded.
```
Why is it breaking my stuff! :)
It's doing it every 60 second interval. 🔥

**Comment by antirotor** [comment]
could it be because we are saving stuff to tmp file when extracting environments and Deadline detects it? I wouldn't dismiss it just because it works with single machine as Deadline might feel it doesn't need to synchronize it then.

One more thing - something touching files in repository share? Like antivirus?

**Comment by BigRoy** [comment]
> could it be because we are saving stuff to tmp file when extracting environments and Deadline detects it? I wouldn't dismiss it just because it works with single machine as Deadline might feel it doesn't need to synchronize it then.

It does actually happen with a single machine too. However - I've also tested a render without the GlobalJobPreLoad doing anything of relevance (basically disabling the full injecting logic) and run it with a single machine. Same issue. I'm going to update to latest Deadline just to be sure ;)

> One more thing - something touching files in repository share? Like antivirus? But then it wouldn't work even with single machine. Damn.

Not as far as I could tell - the files appear static in the folder (the folder on the server that is!); locally however, you might be right that it is a possibility if all machines somehow started doing it.

**Comment by BigRoy** [comment]
> I'm going to update to latest Deadline just to be sure ;)

- Updated Server + Worker to `10.1.20.3 Release  (1a0d97d0f)`
- Disabled all Event Plug-ins
- Removed any custom `GlobalJobPreLoad.py` files
- Cleaned up the Deadline database since it was over 4 GB - removed over 10.000 jobs just to be sure.
- I've confirmed nothing is happening in the "Console" in Deadline at the same time.
- I've confirmed there are no Background Operations running in Deadline at the same time.
- Running the test with a single machine (single machine in Allow List) for the Job.

No luck. Same issue, Deadline force reloads the plug-in every three/four frames at about every 60 second interval. If the next new Task starts >=60 seconds since the last time the Plug-in was "synced" it will trigger the synchronize:

> Synchronizing Plugin MayaBatch from N:\DeadlineRepository10\plugins\MayaBatch took: 0 seconds

After which it will unfortunately reload the plug-in again:

> Plugin will be reloaded: Plugin files are out of date and must be reloaded.

Here is the full Worker log starting with the command "Done executing" from the last Task up to where the new Task starts and starts to reload the Plug-in.

```
2022-01-27 11:25:25:  0: STDOUT: mel: READY FOR INPUT
2022-01-27 11:25:25:  0: Done executing plugin command of type 'Render Task'
2022-01-27 11:25:28:  Skipping pending job scan because it is not required at this time
2022-01-27 11:25:28:  Skipping repository repair because it is not required at this time
2022-01-27 11:25:28:  Skipping house cleaning because it is not required at this time
2022-01-27 11:25:30:  Scheduler Thread - Job's Limit Groups: 
2022-01-27 11:25:30:  All job files are already synchronized
2022-01-27 11:25:30:  Synchronizing Plugin MayaBatch from Z:\NETRENDER\DeadlineRepository10\plugins\MayaBatch took: 0 seconds
2022-01-27 11:25:30:  0: Plugin will be reloaded: Plugin files are out of date and must be reloaded.
```

Here you can clearly see the pattern in the rendertimes that about every minute or so the Plug-in tries sync again and force Plug-in reload.
![deadline_job_reload_plugins_every_sixty_seconds](https://user-images.githubusercontent.com/2439881/151341410-36ad90b7-9d59-4a8b-ac3f-9cf6342cbcc5.png)

---

At this stage this issue seems like a Deadline Support issue - and isn't necessarily OpenPype related. I've posted the question on SysAdminSlack since I've heard some Deadline people hang out there. I'll also e-mail Deadline support however.

**Comment by BigRoy** [comment]
Oh lord. After much more debugging I've found a case where it does not happen and where it does with just a single tweak.

It does not happen in Maya 2020 jobs.
It does happen in Maya 2022 jobs.

Only difference is I submitted the same scene (saved in Maya 2020) as a Maya 2020 job and a Maya 2022 job.

The Maya 2020 job does NOT reload the plug-ins (note that my current Deadline setup has completely reset settings, no GlobalJobPreLoad, no events) whereas the Maya 2022 job does the same "resync" plug-in every 60 second interval.

**Comment by BigRoy** [comment]
Dropping this for anyone else in the future - some things that got recommended to me to try by **Mike Owen** which I thought contained some good bits and tips:

> Logging a ticket to Thinkbox Support is your best option here so we can get to the bottom of this. It's taken quite some time to read that entire OpenPype GitHub issue but a few thoughts (I am going to assume at this point in your debugging that ALL event plugins, Global scripts, etc have all been removed from the equation to help us narrow down this issue):
> 
> 1.  "Reload Plugin Between Tasks" & "Re-synchronize Auxiliary Files Between Tasks" should both be disabled.
> 2.  If any scripts throw an error during the render process, the Worker should reload the entire plugin. You should look to see if there are any error reports on the actual [Worker logs](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/logs.html#worker) instead of relying on the Deadline job reports for the affected jobs. As a test, if you have any pre/post Task scripts defined in your job props, remove delete them and test again.
> 3.  Are all the job timeouts set to default/not-applied as per [here](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/job-controlling.html#timeouts)?
> 4.  Has the job been enabled to be [interrupted](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/job-controlling.html#interruptible)?
> 5.  Have you defined any custom [asset/script job dependencies](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/job-controlling.html#assets) that are being invalidated after the task starts? Remove all to test again.
> 6.  Please take a look at your Deadline Launcher. In the GUI, is this setting enabled? "*Keep Local Repository Cache*" (try disabling it and purge your local "cache" directory) OR are you writing anything into this local cache directory and causing a trigger by the Worker? (I will assume your not using RCS) It might be possible this cache has corrupted or has permissions/access issue. Check state of directories under "cache" and "slave/worker" in your local client. Depending on OS, paths are: <https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/client-config.html#configuration-file-locations>.
> 7.  Last thought. Check NTP on everything. Are all your machines (DL repo, Workers, Pulse, etc) all in the same time-zone and NTP is in sync?
> 8.  Wait for Thinkbox Support to reply to you.
> 
> This "60sec" is definitely the Worker/scheduler thread checking back into the Deadline DB/repo for various reasons, so that bit is well understood. The root cause here is Worker is for whatever reason believing that something has invalidated/modified; which is what we need to root-cause here. In any case, not seen this in 15+yrs of Deadline...so I will initially say it sounds very much "customer-specific" issue here.

--- 

>  Logging a ticket to Thinkbox Support is your best option
> I did earlier Today. The request is `106136`
> 
> 1. Yes, both disabled. :white_check_mark: 
> 2. No errors on this particular Job for that Worker looking at the live Worker Log. :white_check_mark: 
> 3. Yes, all timeouts disabled / not applicable. :white_check_mark: 
> 4. No, not set to be interuptable. :white_check_mark: 
> 5. I did, yes. But have also tested with all dependencies removed. So tested. :white_check_mark: 
> 6. It was disabled on my machine - on which I'm also running the test job. :white_check_mark:  Not sure regarding the RCS - but it's all just regular LAN.
> 7. I actually synced our Deadline Server (Repository host) as it got a time a shift compared to the workstations. How do I actually correctly check whether they'd be "sync enough"? :warning: 
> 8. Thanks. 🚒 

Even reset more bits:

> For sake of testing I just restarted the full Server too. Just to ensure there weren't any lingering services doing anything.
> No luck - problem persists. :fire:
> 
> Then because I'm so damn curious if it does anything I've went ahead and clicked "Reset Settings" for the full Deadline Repository Options.
> No luck - problem persists. :fire:
> 
> After which I just killed all local deadline processes (launcher, worker, etc.) and restarted again just to ensure it uses latest up-to-date settings.
> No luck - problem persists. :fire:

**Comment by BigRoy** [comment]
I found the Maya 2022 issue. :)

It's due to Maya 2022 being Python 3 which creates __pycache__ folders by default for code that it runs. Since it's running the code in the copied plug-in files on the local disk it is creating folders. It is Deadline detecting those changes that it feels the need to resync. And voila.

The issue is completely gone when I set environment variable `PYTHONDONTWRITEBYTECODE=1` for the Job. That disables Python 3 from creating those Pycache files/folders. I guess until a fix is out for Deadline I might just as well set that environment variable in `GlobalJobPreLoad`? Thoughts?

**Comment by BigRoy** [comment]
I'm closing this for now since our particular issue has been resolved since:

- Reducing time taken to connect to mongodb by resolving hostname -> IP address beforehand (likely this was a DHCP issue on our network) - which saved a lot of time on our end
- Working around MayaBatch Plug-in restarting every 60s or per frame for Maya 2022 [as reported in this comment](https://github.com/pypeclub/OpenPype/issues/2592#issuecomment-1023748942)
- We also disabled OPENPYPE_VALIDATE_VERSION on the farm which performs checksums of the files just because it helped a tiny bit - I wouldn't necessarily recommend this is if you have very big infrastructure since it might make your life harder to debug an edge case where files might've actually gone corrupt inside OpenPype. But for us it was a 'likely never' scenario that outweighed the average decrease in speed.
- Additionally #2607 also got merged which helped some more.
